### PR TITLE
[Homepage] initial translation

### PR DESCRIPTION
**Pull request by itszero**
Taking the first stab at the homepage!

We haven't got an official glossary list yet so I'm taking some liberty here. It's mostly straightforward but I decided to rewrite a few sentences to make it more readable. (I prefer to make the translation more readable/smooth rather than stick to the original structure)

I also somewhat localized the examples code a bit so it feels more approachable.

A few comments:

- ~~*hooks* was used in the external plugins example. I removed it to prevent confusion with the React Hooks (and also that word is tricky to translate into TC, I think broader Taiwanese programming community, as in outside of React like Windows MFC community, mostly just use the word hook directly so I'd like to avoid that confusion.)~~ nvm lol. Pulled the master and found out that upstream made the same change too.
- Going with a bit more localized terms than English terms -- happy to tweak the ratio here!
- Translated header/footer too

### CODE REVIEW

File: src/pages/index.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ class Home extends Component {
     return (
       <Layout location={location}>
         <TitleAndMetaTags
-          title="React &ndash; A JavaScript library for building user interfaces"
+          title="React &ndash; 用來實做使用者界面的 JavaScript 函式庫"
```
Comment (neighborhood999) [inline src/pages/index.js]: I think `實做` should be `實作`.

File: src/pages/index.js [hunk 2]
```diff
@@ -134,7 +134,7 @@ class Home extends Component {
                         fontSize: 30,
                       },
                     }}>
-                    A JavaScript library for building user interfaces
+                    用來實做使用者界面的 JavaScript 函式庫
```
Comment (neighborhood999) [inline src/pages/index.js]: Same above.

File: content/home/examples/a-simple-component.js [hunk 3]
```diff
@@ -2,13 +2,13 @@ class HelloMessage extends React.Component {
   render() {
     return (
       <div>
-        Hello {this.props.name}
+        吃飽沒, {this.props.name}？
```
Comment (neighborhood999) [inline content/home/examples/a-simple-component.js]:
```suggestion
        吃飽沒，{this.props.name}？
```

File: content/home/examples/a-simple-component.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
 ---
-title: A Simple Component
+title: 簡單的 Component
 order: 0
 domid: hello-example
 ---
 
-React components implement a `render()` method that takes input data and returns what to display. This example uses an XML-like syntax called JSX. Input data that is passed into the component can be accessed by `render()` via `this.props`.
+React component 需要實做 `render()` function，這個函式接受輸入的資料並回傳需要顯示的內容。這個範例使用了 JSX，一個類似 XML 的語法。傳入給元件的輸入資料可以透過 `this.props` 在 `render()` function 中存取。
```
Comment (neighborhood999) [inline content/home/examples/a-simple-component.md]:
```suggestion
React component 需要實做 `render()` function，這個 function 接受輸入的資料並回傳需要顯示的內容。這個範例使用了 JSX，一個類似 XML 的語法。傳入給元件的輸入資料可以透過 `this.props` 在 `render()` function 中存取。
```

File: content/home/examples/an-application.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
-title: An Application
+title: 一個應用程式
 order: 2
 domid: todos-example
 ---
 
-Using `props` and `state`, we can put together a small Todo application. This example uses `state` to track the current list of items as well as the text that the user has entered. Although event handlers appear to be rendered inline, they will be collected and implemented using event delegation.
+透過靈活運用 `props` 跟 `state`，我們可以寫一個很簡易的待辦事項 App。這個範例使用 `state` 保存目前所有的待辦事項以及使用者目前輸入的文字。雖然這些 event handler 像是直接寫在程式碼中，實際上 React 內部會自動收集他們並透過 event delegation 實做。
```
Comment (neighborhood999) [inline content/home/examples/an-application.md]:
```suggestion
透過靈活運用 `props` 跟 `state`，我們可以寫一個很簡易的待辦事項的應用程式。這個範例使用 `state` 保存目前所有的待辦事項以及使用者目前輸入的文字。雖然這些 event handler 像是直接寫在程式碼中，實際上 React 內部會自動收集他們並透過 event delegation 實做。
```

File: content/home/marketing/component-based.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ---
-title: Component-Based
+title: 元件式開發
 order: 1
 ---
 
-Build encapsulated components that manage their own state, then compose them to make complex UIs.
+首先實做一個自我獨立的 component，然後活用他們建立複雜的使用者界面。
```
Comment (neighborhood999) [inline content/home/marketing/component-based.md]:
```suggestion
首先實作一個擁有 state 的獨立 component，然後組合他們建立複雜的使用者介面。
```

File: content/home/marketing/declarative.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
 ---
-title: Declarative
+title: 宣告式
 order: 0
 ---
 
-React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes.
+React 讓實做互動式的使用者界面變得一點也不痛苦。 你只要在你的應用程式中為每個情境設計一個簡單的 View ，React 就為在資料變更時自動有效率的只更新並重繪有異動的元件。
```
Comment (neighborhood999) [inline content/home/marketing/declarative.md]:
```suggestion
React 讓實作互動式的使用者界面變得一點也不痛苦。你只要在你的應用程式中為每個情境設計一個簡單的 view，React 就會在資料變更時有效率的自動更新並 render 有異動的元件。
```

File: content/home/marketing/declarative.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
...
 
-Declarative views make your code more predictable and easier to debug.
+宣告式的 View 讓你更容易預測你的程式的行為，同時也較為容易除錯。
```
Comment (neighborhood999) [inline content/home/marketing/declarative.md]: `View` -> `view`

File: content/home/examples/a-simple-component.js (hunk 3 above)
Comment (itszero) [reply content/home/examples/a-simple-component.js]: reverting code example localization per discucsions

**Comment by itszero** [comment]
Thanks for the review. I think i've updated everything!

**Pull request merged**